src/decorators/cache/index.js

```diff
@@ -0,0 +1,81 @@
+import isNode from '../../helpers/isNode'
+import md5 from '../../helpers/md5'
+import stringOrIntToMs from '../../helpers/stringOrIntToMs'
+
+import LRU from './algorithms/LRU'
+import LFU from './algorithms/LFU'
+
+import Tracker from './tracker'
+
+const DEFAULT_TTL = 500
+
+const isPromise = (obj) => typeof obj !== 'undefined' &&
+  typeof obj.then === 'function'
+
+const _cache = ({ttl, target, name, instance, original, server, algorithm, host, size} = {}) => {
+  const cache = algorithm === 'lru' ? new LRU({size})
```

Comment:
I think this line does this code not very scalable.
It'd be great to have an Abstract Factory to recover the right Cache class.

const cache = FactoryCache.create(algorithm, {size})

The abstract factory could be something like

class FactoryCache 
{
    constructor() {
      this._registeredTypes = [{ name: 'lfu', class: LFU }, { name: 'lru', class: LRU}]
    }

    create(algorithm, {size}) {
        const registeredType = this._registeredTypes.find(current => current.name === algorithm)

        return registeredType != null 
            ? new registeredType.class({size})
            : new Error(`[cv-decorators::cache] unknow algorithm: ${algorithm}`)
    }
}